week4/json-print/jasonkuhrt/Main.hs

```diff
@@ -0,0 +1,119 @@
+{- README
+
+-- Code Kata : JSON Print
+
+Define a data type for JSON and print it as JSON (without indentation).
+Don't forget that you should support floating-point numbers and escaping in
+strings.
+-}
+module Main where
+
+import Data.Fixed (mod')
+import Data.Map (Map)
+import qualified Data.Map as Map
+import qualified Data.List as List
+
+
+
+data JSON =
+    Nil
+  | B Bool
+  | S String
+  | N Double
+  | L [JSON]
+  | O (Map String JSON)
+  deriving (Show)
+
+
+
+{- | Stringify a JSON type to a valid JSON string. -}
+stringify :: JSON -> String
+
+stringify Nil        = "null"
+
+stringify (B True)   = "true"
+stringify (B False)  = "false"
+
+stringify (S string) =
+    wrapQuotes
+  . escape '\\' -- Otherwise parse yields invalid/different strings. [1]
```

Comment:
Makes sense. The revised solution changes all this anyways since as you said this was failing to escape control characters.